Batteries/Util/ProofWanted.lean

```diff
@@ -33,7 +33,10 @@ elaboration, but it's then removed from the environment.
 def elabProofWanted : CommandElab
   | `($mods:declModifiers proof_wanted $name $args* : $res) => withoutModifyingEnv do
     -- The helper axiom is used instead of `sorry` to avoid spurious warnings
-    elabCommand <| ← `(axiom helper (p : Prop) : p
-                       $mods:declModifiers
-                       theorem $name $args* : $res := helper _)
+    elabCommand <| ← `(
+      section
+      set_option linter.unusedVariables false
+      axiom helper {α : Sort _} : α
```

Comment:
Do we really want `Sort _` here?

Reply:
Ah, I see this incorporates `definition_wanted`.

Reply:
Yes, I found that the error messages for non-theorems was unfriendly for just Prop. The test file shows that non-Prop still gives a good error.